### PR TITLE
Add publish.yml to publish to PyPi (#30)

### PR DESCRIPTION
**Pull request by quasi-mod**
This patch adds publish.yml to enable the workflow to publish to PyPi.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    environment:
+      name: pypi
+      url: https://pypi.org/p/pywebsocket3
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+    steps:
+    # retrieve your distributions here
+
+    - name: Publish package distributions to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
```
Comment (ricea) [inline .github/workflows/publish.yml:14]: I guess `release/v1` needs to be replaced with an actual tag in our repository? It should probably match the version number in `setup.py`.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/pywebsocket3
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+    steps:
+    # retrieve your distributions here
```
Comment (ricea) [inline .github/workflows/publish.yml:11]: This comment is confusing to me, but maybe we need a download step like here: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#publishing-the-distribution-to-pypi ?

File: .github/workflows/publish.yml (hunk 1 above)
Comment (quasi-mod) [reply .github/workflows/publish.yml:30]: I think this version is to specify the version for github action `pypa/gh-action-pypi-publish`, so I don't think it's for pywebsocket's version

File: .github/workflows/publish.yml (hunk 2 above)
Comment (quasi-mod) [reply .github/workflows/publish.yml]: Seemed like that is the case. I have added a new set of steps to build distribution package.

**Comment by quasi-mod** [comment]
Thanks for the comment. I have update the patch. Please take a look.

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    # Install required dependencies
+    - name: Install dependencies
+      run: |
+        python3 -m pip install --upgrade pip
+        python3 -m pip install six yapf
+
+    # Build and create dist
+    - name: Build Package
+      run: python3 setup.py sdist
```
Comment (ricea) [inline .github/workflows/publish.yml:26]: We should probably migrate to `python -m build`, but there's no need to do that now. I don't if it works with Python 2.7, anyway.

**Review by ricea — APPROVED** [review]
lgtm

File: .github/workflows/publish.yml (hunk 3 above)
Comment (quasi-mod) [reply .github/workflows/publish.yml:26]: Sounds reasonable.

**Pull request merged**